src/store/reader.rs

```diff
@@ -93,6 +93,70 @@ impl StoreReader {
     }
 }
 
+pub struct StoreReaderIterator<'a> {
+    block_offset: usize,
+    document_offset: usize,
+    store_reader: &'a StoreReader,
+    uncompressed_block: Vec<u8>,
+}
+
+impl<'a> StoreReaderIterator<'a> {
+    pub fn has_more_documents(&self) -> bool {
+        self.document_offset < self.uncompressed_block.len()
+    }
+
+    pub fn has_more_blocks(&self) -> bool {
+        self.block_offset < self.store_reader.block_data().len()
+    }
+
+    fn read_block(&mut self) -> crate::Result<()> {
+        let (block_size, compressed_block) = self.store_reader.compressed_block(self.block_offset);
+
+        self.uncompressed_block.clear();
+        decompress(&compressed_block, &mut self.uncompressed_block)?;
+
+        self.block_offset += block_size;
+        self.document_offset = 0;
+        Ok(())
+    }
+}
+
+impl<'a> IntoIterator for &'a StoreReader {
```

Comment:
This Iterator does not take in account deletes.

If you have a need for this, we can keep it, but let's make it explicit. In that case let's remove the `IntoIterator` implementation, and add a method `iter_docs_including_deleted()` with some documentation explaining this behavior.

We can also have an `iter_docs(&DeleteBitset)` but not that the block decompression would have to be made lazy in that case.

Reply:
Imo the method must take deletion into account for the logical consistency. I'm totally missed this point. So, should I leave the iterator interface and add there DeleteBitset checking?